File: utils/req.go
package utils

import (
	"fmt"
	"io"
	"net/http"
	"os"
	"strings"
)

func MakeRequest(day int) string {
	url := fmt.Sprintf("https://adventofcode.com/2023/day/%v/input", day)

	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		panic(err)
	}

	req.AddCookie(
		&http.Cookie{
			Name:  "session",
			Value: os.Getenv("AOC_SESSION"),
		},
	)

	client := http.Client{}
	response, err := client.Do(req)
	if err != nil {
		panic(err)
	}

	body, err := io.ReadAll(response.Body)
	if err != nil {
		panic(err)
	}

	return string(body)
}

func FormatInput(input string) []string {
	return strings.Split(strings.TrimSuffix(input, "\n"), "\n")
}

func FormattedRequest(day int) []string {
	return FormatInput(MakeRequest(day))
}

File: pythonutils/req.py
import os
import requests


def make_request(day: int) -> str:
    session_cookie = os.getenv("AOC_SESSION")
    if not session_cookie:
        raise ValueError("AOC_SESSION environment variable must be set")

    url = f"https://adventofcode.com/2023/day/{day}/input"

    cookies = {"session": session_cookie}
    response = requests.get(url, cookies=cookies)
    response.raise_for_status()

    return response.text


def format_input(input_str: str) -> list[str]:
    return input_str.rstrip('\n').split('\n')


def formatted_request(day: int) -> list[str]:
    return format_input(make_request(day))
